src/sage/combinat/matrices/hadamard_matrix.py

```diff
@@ -1728,25 +1736,35 @@ def hadamard_matrix(n, existence=False, check=True):
         sage: matrix.hadamard(324, existence=True)
         True
     """
+    name = str(n)
+    if construction_name:
+        def report_name(nam):
+            return nam
+    else:
+        def report_name(nam):
+            return True
+
     if not (n % 4 == 0) and (n > 2):
         if existence:
             return False
         raise ValueError("The Hadamard matrix of order %s does not exist" % n)
     if n == 2:
         if existence:
-            return True
+            return report_name(name)
         M = matrix([[1, 1], [1, -1]])
     elif n == 1:
         if existence:
-            return True
+            return report_name(name)
         M = matrix([1])
     elif is_prime_power(n//2 - 1) and (n//2 - 1) % 4 == 1:
+        name = "paleyII " + name
```

Comment:
why setting this variable that is not reused ? you could put the string construction directly in the call of the function.

Reply:
There is a bit of a design deficiency in this code, which I am not going to address here - the matrices are just Sage matrices, you cannot attach names to them. A better way would be to have a derived class, I suppose. But not now. This is actually a quick fix to please a referee of a paper we wrote about these constructions.

Reply:
I understand the motivation for this quick fix.

You do `name = "paleyI" + name` and then, if `existence` is `True`, you call `report_name(name)`. If `existence` is `False`, the modified name is not used in the method. So you could simply call `report_name("paleyI" + name)` and avoid `name = "paleyI" + name`.

Reply:
I'd like to keep `name` as it is, as I plan to have more detailed name reporting, for recursive constructions, implemented,  as well. So `name` will get more "parents" info...

Reply:
ok.